pkg/apis/maistra/v2/extension_providers.go

```diff
@@ -52,20 +55,44 @@ type ExtensionProviderEnvoyExternalAuthorizationHTTPConfig struct {
 	IncludeRequestHeadersInCheck []string `json:"includeRequestHeadersInCheck,omitempty"`
 	// Set of additional fixed headers that should be included in the authorization request sent to the authorization service.
 	// Key is the header name and value is the header value.
-	// Note that client request of the same key or headers specified in include_request_headers_in_check will be overridden.
+	// Note that client request of the same key or headers specified in includeRequestHeadersInCheck will be overridden.
 	IncludeAdditionalHeadersInCheck map[string]string `json:"includeAdditionalHeadersInCheck,omitempty"`
 	// If set, the client request body will be included in the authorization request sent to the authorization service.
 	IncludeRequestBodyInCheck *ExtensionProviderEnvoyExternalAuthorizationRequestBodyConfig `json:"includeRequestBodyInCheck,omitempty"`
 }
+type ExtensionProviderEnvoyExternalAuthorizationGRPCConfig struct {
+	// REQUIRED. Specifies the service that implements the Envoy ext_authz gRPC authorization service.
+	// The format is `[<Namespace>/]<Hostname>`. The specification of `<Namespace>` is required only when it is insufficient
+	// to unambiguously resolve a service in the service registry. The `<Hostname>` is a fully qualified host name of a
+	// service defined by the Kubernetes service or ServiceEntry.
+	//
+	// Example: "my-ext-authz.foo.svc.cluster.local" or "bar/my-ext-authz.example.com".
+	Service string `json:"service"`
+	// REQUIRED. Specifies the port of the service.
+	Port int64 `json:"port"`
+	// The maximum duration that the proxy will wait for a response from the provider, this is the timeout for a specific request (default timeout: 600s).
+	// When this timeout condition is met, the proxy marks the communication to the authorization service as failure.
+	// In this situation, the response sent back to the client will depend on the configured `failOpen` field.
+	Timeout *string `json:"timeout,omitempty"`
+	// If true, the HTTP request or TCP connection will be allowed even if the communication with the authorization service has failed,
+	// or if the authorization service has returned a HTTP 5xx error.
+	// Default is false. For HTTP request, it will be rejected with 403 (HTTP Forbidden). For TCP connection, it will be closed immediately.
+	FailOpen bool `json:"failOpen,omitempty"`
```

Comment:
I noticed there were pointers @jewertow had used, is this fine to not use a pointer (following upstream). Or is there a reason for us to change it to a pointer?

Reply:
We use pointers in API structures to know whether a particular field was specified by a user or not. In case of value fields, you don't know whether `failOpen: false` comes from user input or is it just the default bool value.
However, we only need to know that for required fields. Optional fields don't have to be pointers.

Reply:
I was wrong, I'm sorry. We use pointers for optional values to know whether we should set them in helm values or not.

Reply:
But now I see that I didn't check default values in `validateExtensionProviders` for required fields, i.e. service and port. We can proceed this as is and improve that in a follow-up PR.

Reply:
So the following fields should be pointers: timeout, failOpen, statusOnError, includeRequestBodyInCheck.

Reply:
> to know whether we should set them in helm values or not.

Just for clarification: of course, we might use value fields and then set default values in helm templates, but then we would have to make sure that the default struct values are default API values. For example, default value for boolean field in Go is false, but an API might be implemented so that when it's not specified, then it's set to true.

Reply:
I changed failOpen and statusOnError back to pointers. includeRequestBodyInCheck was already one.